Added setting to allow response methods to be any module, not just ajax.py

Ghopper21 opened the pull request:
Added a setting DAJAXICE_IMPLICIT_MODULE_NAME which lets you choose what (if any) implicit Python module name will be added to calls to response methods via Dajaxice. Specifically, if DAJAX_IMPLICIT_MODULE_NAME is implicit_module, a Javascript call to Dajaxice.app_name.method_name will cause Dajaxince to send the request to app_name.implicit_module.method_name  

In other words, with this setting you can put response methods in any module, not just in ajax.py.

By default, this setting is "ajax" -- which leaves the existing functionality unchanged, i.e. response methods go in ajax.py files within app packages. Note this setting can be a blank string, which allows the Javascript Dajaxice call to mirror the Python package/module/method path more closely.

Also note that you should still always register Dajaxice response methods in a Django app's ajax.py module (even if the response methods themselves are not in that module). This is because Dajaxice's autodiscover mechanism will always look the ajax.py moduleregardless of this setting.

Ghopper21 committed:
Added settings.DAJAXICE_IMPLICIT_MODULE_NAME functionality

Ghopper21 committed:
Changed accidental tabs to spaces

Ghopper21 committed:
Hide 'something goes wrong' pop-up error caused by timing

fonfon commented:
I'm not a developer of dajaxice, and can't accept/merge anything here - but I'm having the same trouble at the moment. It's limiting to be forced to use 'app'/ajax.py.

But I don't like the idea of an additional setting either, this would be equally limiting for me.

Perhaps your functionality is possible without an additional setting? You would have to change dajaxice_register decorator, get out where the function you are decorating is located (relative to project root), and use this information later on, instead of your new settings entry.

seawaywen committed:
tweaks to work with django1.6

seawaywen committed:
tweaks to work with django1.6

seawaywen committed:
tweaks to work with django1.6

seawaywen committed:
fixed the issue that still use simplejson after import the json package

Commit:
Merge branch 'master' of https://github.com/jorgebastida/django-dajaxice

Conflicts:
	dajaxice/core/DajaxiceRequest.py

Commit:
Change an import for update to Django 1.6.

Ghopper21 committed:
Merge pull request #1 from azlyth/master

Update to present and also update an import (to enable use for Django 1.6).

Commit:
Merge branch 'master' of https://github.com/seawaywen/django-dajaxice

Ghopper21 committed:
Merge pull request #2 from azlyth/master

Additional Django 1.6 tweaks (update simplejson to json)

airikh committed:
Make compatible with Django 1.7+

`mimetype` keyword was removed from HttpResponse in Django 1.7; `content_type` should be used
(see https://docs.djangoproject.com/en/1.7/internals/deprecation/#deprecation-removed-in-1-7)

Ghopper21 committed:
Merge pull request #3 from Ghopper21/django-1.7-support

Make compatible with Django 1.7+

airikh committed:
Add missing django import

Ghopper21 committed:
Merge pull request #4 from Ghopper21/bugfix-missing-django-import

Add missing django import

korsunowk committed:
Change django utils to standard importlib library

Ghopper21 committed:
Merge pull request #5 from Ghopper21/managerjellyfish/ch229/fix-use-of-django-utils-importlib-in-dajaxice

Fix use of django utils importlib in dajaxice

korsunowk committed:
Remove deprecated in Django 1.9 url from future, use build-in url

Ghopper21 committed:
Merge pull request #6 from Ghopper21/managerjellyfish/ch288/fix-dajaxice-problems-with-django-1-9

Remove deprecated in Django 1.9 url from future, use build-in url

korsunowk committed:
Fix all things for upgrade to the Django 1.9 version

Ghopper21 committed:
Merge pull request #7 from Ghopper21/managerjellyfish/ch288/fix-dajaxice-problems-with-django-1-9

Fix all things for upgrade to the Django 1.9 version
